Cloudflare/gateway_http/ingest/parser.yml

```diff
@@ -0,0 +1,94 @@
+name: cloudflare-gateway-http
+pipeline:
+  - name: json_event
+    external:
+      name: json.parse-json
+      properties:
+        input_field: "{{original.message}}"
+  - name: set_ecs_fields
+  - name: set_cloudflare_fields
+stages:
+  set_ecs_fields:
+    actions:
+      - set:
+          "@timestamp": '{{json_event.message.Datetime}}'
+          destination.ip: '{{json_event.message.DestinationIP}}'
+          destination.port: '{{json_event.message.DestinationPort}}'
+          device.id: '{{json_event.message.DeviceID}}'
+          event.action: '{{json_event.message.Action}}'
+          event.category: ["network"]
+          event.dataset: "gateway_http"
+          event.kind: 'event'
+          event.type: ["info"]
+      - set:
+          event.type: ["info", "allowed"]
+        filter: '{{json_event.message.Action == "allow"}}'
+      - set:
+          event.type: ["denied"]
+        filter: '{{json_event.message.Action == "deny"}}'
+      - set:
+          http.request.method: '{{json_event.message.HTTPMethod}}'
+          http.request.referrer: '{{json_event.message.Referer}}'
+          http.response.status_code: '{{json_event.message.HTTPStatusCode}}'
+          http.version: '{{json_event.message.HTTPVersion}}'
+          network.protocol: 'http'
+          observer.type: "proxy"
+          observer.vendor: "Cloudflare"
+          source.ip: '{{json_event.message.SourceIP}}'
+          source.port: '{{json_event.message.SourcePort}}'
+          user_agent.original: '{{json_event.UserAgent}}'
+          user.id: '{{json_event.message.UserID}}'
+          user.email: '{{json_event.message.Email}}'
+          url.original: '{{json_event.URL}}'
+
+  set_cloudflare_fields:
+    actions:
+      - set:
+          cloudflare.AccountID: '{{json_event.message.AccountID}}'
+        filter: '{{json_event.message.AccountID is not in [""]}}'
+      - set:
+          cloudflare.BlockedFileHash: '{{json_event.message.BlockedFileHash}}'
+        filter: '{{json_event.message.BlockedFileHash is not in [""]}}'
+      - set:
+          cloudflare.BlockedFileName: '{{json_event.message.BlockedFileName}}'
+        filter: '{{json_event.message.BlockedFileName is not in [""]}}'
+      - set:
+          cloudflare.BlockedFileReason: '{{json_event.message.BlockedFileReason}}'
+        filter: '{{json_event.message.BlockedFileReason is not in [""]}}'
+      - set:
+          cloudflare.BlockedFileSize: '{{json_event.message.BlockedFileSize}}'
+        filter: '{{json_event.message.BlockedFileSize is not in [""]}}'
+      - set:
+          cloudflare.BlockedFileType: '{{json_event.message.BlockedFileType}}'
+        filter: '{{json_event.message.BlockedFileType is not in [""]}}'
```

Comment:
Except `BlockedFileReason`, I wonder if `file.*` is not suitable for this information?

Reply:
I changed this, except for `BlockedFileHash ` because we do not know the type of hash to associate it with.